src/components/LoginSignup/SignUp.js

```diff
@@ -0,0 +1,133 @@
+import React from "react";
+import { useState } from "react";
+import { navigate } from "@reach/router";
+import { signUp } from "../../auth/signUp";
+import { validate as isEmail } from "isemail";
+import { makeStyles } from "@material-ui/core/styles";
+import { TextField, Button } from "@material-ui/core/";
+import { Wrapper } from "./Wrapper";
+
+const useStyles = makeStyles((theme) => ({
+  textField: {
+    marginBottom: theme.spacing(2),
+    "& label.Mui-focused": {
+      color: theme.palette.grey[900],
+    },
+  },
+  button: {
+    marginTop: theme.spacing(2),
+    color: theme.palette.grey[900],
+  },
+  link: {
+    textDecoration: "none",
+    color: theme.palette.grey[900],
+  },
+  errorMessage: {
+    marginBottom: theme.spacing(2),
+    color: theme.palette.error.main,
+  },
+}));
+
+function SignUp({ location }) {
+  const classes = useStyles();
+  const [emailValue, setEmailValue] = useState("");
+  const [passwordValue, setPasswordValue] = useState("");
+  const [usernameValue, setUsernameValue] = useState("");
+  const [confirmPassword, setConfirmPassword] = useState("");
+  const [errorMessage, setErrorMessage] = useState("");
+
+  const onSignUpClicked = async () => {
+    setErrorMessage("");
+    const validationError = validateForm();
+    if (validationError) {
+      setErrorMessage(validationError);
+      return;
+    }
+
+    try {
+      const signUpSuccessful = await signUp(usernameValue, emailValue, passwordValue);
+      signUpSuccessful && navigate(`/login`, { state: { signUpSuccessful } });
+    } catch (e) {
+      switch (e.message) {
+        case "auth/email-already-in-use":
+          setErrorMessage("Die angegebene E-Mail Adresse wird bereits verwendet.");
+          break;
+        case "auth/weak-password":
+          setErrorMessage("Dein Passwort muss mehr als sechs Zeichen haben.");
```

Comment:
```suggestion
          setErrorMessage("Dein Passwort muss mindestens 6 Zeichen haben.");
```
6 ist einfacher zu lesen als sechs und genügt man braucht nicht mehr als 6 😉